node_cli/realm/realmTasks.js

```diff
@@ -0,0 +1,101 @@
+var Realm = require("realm");
+var mongodb = require("mongodb");
+const schemas = require("./schemas");
+const index = require("../index");
+
+async function setRealm(user) {
+  const config = {
+    schema: [schemas.TaskSchema, schemas.UserSchema, schemas.ProjectSchema],
+    sync: {
+      user: user,
+      partitionValue: "myPartition",
+    },
+  };
+
+  return await Realm.open(config).then((r) => {
+    return r;
+  });
+}
+
+async function getTasks(user) {
+  return setRealm(user)
+    .then(async (r) => {
+      let allTasks = await r.objects("Task");
+      return allTasks;
+    })
+    .catch((err) => {
+      return index.output(err, "error");
+    });
+}
```

Comment:
This is an `async` function so you can `await` instead of using the promise chain. Same with the others in this file.

```js
async function getTasks(user) {
  try {
    const realm = await setRealm(user);
    const allTasks = await r.objects("Task");
    return allTasks;
  } catch(err) {
    return index.output(err, "error");
  }
})
```

We may also consider leaving off the error handling here (or throwing a custom error) and then handling errors in the function where we call these methods.